src/components/ModalWithForm.js

```diff
@@ -6,6 +6,18 @@ export default class ModalWithForm extends Modal {
     this._modalForm = this._modalElement.querySelector(".modal__form");
     this._inputList = this._modalForm.querySelectorAll(".modal__input");
     this._handleFormSubmit = handleFormSubmit;
+    this._modalButton = this._modalForm.querySelector(".modal__button");
+    this._modalButtonText = this._modalButton.textContent;
+  }
+
+  setModalLoad(isLoading) {
+    if (isLoading) {
+      this._modalButton.textContent = "Saving...";
```

Comment:
**Could be improved**

`"Saving..."` is a hardcode here

It’s better to make a `renderLoading` method inside `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```